src/components/apps/launch/MultiInputSelector.js

```diff
@@ -0,0 +1,324 @@
+
+    const paths = getIn(values, name);
+    const errorMsg = getFormError(name, touched, errors);
+
+    const tableId = buildDebugId(id, "table");
+    const tableLabelID = buildDebugId(tableId, "tableLabel");
+
+    const onRequestSort = (event, property) => {
+        const newOrder = order === "desc" ? "asc" : "desc";
+        setOrder(newOrder);
+        setFieldValue(name, getSortedPaths(newOrder, paths));
+    };
+
+    const onSelectAllClick = (event, checked) => {
+        const selectAll = checked && !selected.length;
+        setSelected(selectAll ? paths : []);
+        setLastSelectedIndex(-1);
+    };
+
+    const onRowSelected = (event, path, index) => {
+        if (event.shiftKey && lastSelectedIndex !== index) {
+            lastSelectedIndex > index
+                ? rangeSelect(index, lastSelectedIndex, path)
+                : rangeSelect(lastSelectedIndex, index, path);
+        } else {
+            toggleSelection(path);
+        }
+
+        setLastSelectedIndex(index);
+    };
+
+    const rangeSelect = (start, end, path) => {
+        const newSelectionRange = paths.slice(start, end + 1);
+        if (selected.includes(path)) {
+            deselect(newSelectionRange);
+        } else {
+            select(newSelectionRange);
+        }
+    };
+
+    const select = (paths) => {
+        setSelected([...new Set([...selected, ...paths])]);
+    };
+
+    const deselect = (paths) => {
+        setSelected(selected.filter((path) => !paths.includes(path)));
+    };
+
+    const toggleSelection = (path) => {
+        if (selected.includes(path)) {
+            deselect([path]);
+        } else {
+            select([path]);
+        }
+    };
+
+    return (
+        <FieldArray name={name}>
+            {(arrayHelpers) => {
+                return (
+                    <>
+                        <Toolbar disableGutters>
+                            <FormControl fullWidth error={!!errorMsg}>
+                                <FormLabel
+                                    id={tableLabelID}
+                                    required={required}
+                                >
+                                    {label}
+                                </FormLabel>
+                                <FormHelperText>{errorMsg}</FormHelperText>
+                            </FormControl>
+                            {!!selected.length && (
+                                <Button
+                                    variant="contained"
+                                    size="small"
+                                    startIcon={<ClearIcon />}
+                                    onClick={() => {
+                                        setSelected([]);
+
+                                        const filteredPaths = paths.filter(
+                                            (path) => !selected.includes(path)
+                                        );
+
+                                        setFieldValue(
```

Comment:
I'm wondering if using `setFieldValue` here may be causing the validation/touched issues described in the main PR description.

I'm tempted to remove support for range-select and the `Remove` selected button, since the user can remove items individually, and if they need a large number of input paths then they could use a multi-input path list file; but then I think we will run into this same issue for the  multi-input path list creator component 🤔